examples/twittermap/web/app/actor/TwitterMapPigeon.scala

```diff
@@ -0,0 +1,43 @@
+package actor
+
+import java.net.URI
+
+import akka.actor._
+import akka.stream.Materializer
+import org.eclipse.jetty.websocket.client.{ClientUpgradeRequest, WebSocketClient}
+import play.api.libs.json.JsValue
+import socket.TwitterMapClientSocket
+
+import scala.concurrent.ExecutionContext
+
+class TwitterMapPigeon (val cloudberryWS: String,
+                        val out: ActorRef)
+                       (implicit ec: ExecutionContext, implicit val materializer: Materializer) extends Actor with ActorLogging {
+
+  val twitterMapWebSocketClient: WebSocketClient = new WebSocketClient
+  val twitterMapClientSocket: TwitterMapClientSocket = new TwitterMapClientSocket(out)
+
+  override def preStart(): Unit = {
+    super.preStart
+    twitterMapWebSocketClient.start()
+    twitterMapWebSocketClient.connect(twitterMapClientSocket, new URI(cloudberryWS), new ClientUpgradeRequest)
```

Comment:
what is `ClientUpgradeRequest`?

Reply:
Since websocket connection begins with an upgrade request in Http protocol, `clientUpgradeRequest` serves for that. I checked the source code of websocket client that it actually can be absent from parameters, which will be internally set up. So I removed it in the updated commits.

Details here:
https://www.eclipse.org/jetty/javadoc/current/org/eclipse/jetty/websocket/client/WebSocketClient.html